### DISCUSSION TITLE
Web3Auth with Auth0

### DISCUSSION THREAD

**Discussion by exhaledev** [Web3Auth/Web3Auth#886] [Help]
Hi All, 

I am trying to integrate web3auth with auth0

In Auth Dashboard I have added https://beta.openlogin.com/auth as the allowed callback url
but i keep getting this error, while im trying to login

<img width="654" alt="image" src="https://user-images.githubusercontent.com/105014422/202215691-8b49a1ec-b1a2-41fb-b89a-97d43a0597b9.png">

<img width="761" alt="image" src="https://user-images.githubusercontent.com/105014422/202215850-5ddc1b58-d445-4c95-ad20-27b9a284c331.png">

My package.json is as follows

   "@alch/alchemy-web3": "^1.4.2",
    "@web3auth/torus-wallet-connector-plugin": "^1.0.0",
    "@web3auth/base": "^3.0.0",
    "@web3auth/modal": "^3.0.3",
    "@web3auth/openlogin-adapter": "^3.0.3",
    "web3": "^1.7.0",
    "axios": "^1.1.3",
    "ethers": "^5.0.0",
    "faunadb": "^4.4.1",
    "next": "^12.1.6",
    "next-auth": "^4.14.0",
    "nextjs-progressbar": "^0.0.14",
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "react-progressbar-line": "^1.0.2",
    "sweetalert2": "^11.4.8",
    "wagmi": "^0.6.0"

And my web3auth login button looks as follows

<img width="581" alt="image" src="https://user-images.githubusercontent.com/105014422/202217449-f073fe9a-8d2f-4b08-ba34-25fb5d3aafb0.png">

**Comment by shahbaz17** [comment]
Hey @exhaledev 

Can you share the name of the verifier?

Also, what are you passing in clientId of connectTo?

**Reply by shahbaz17** [reply]
Also, in your `package.json` file, I don't see the `@web3auth/core` package.

Please look at [this guide](https://web3auth.io/docs/guides/auth0) and [this example](https://github.com/Web3Auth/examples/tree/main/web-core-sdk/custom-authentication/auth0-react-core-example) to see how it's implemented.

**Reply by exhaledev** [reply]
> Hey @exhaledev
> 
> Can you share the name of the verifier?
> 
> Also, what are you passing in clientId of connectTo?

Yeah sure, here is a screenshot below
<img width="571" alt="image" src="https://user-images.githubusercontent.com/105014422/202238654-93b6843a-5b4f-499a-9dc3-8469d268faf7.png">

**Reply by exhaledev** [reply]
@shahbaz17 
> Also, in your `package.json` file, I don't see the `@web3auth/core` package.
> 
> Please look at [this guide](https://web3auth.io/docs/guides/auth0) and [this example](https://github.com/Web3Auth/examples/tree/main/web-core-sdk/custom-authentication/auth0-react-core-example) to see how it's implemented.

I have used that example from the github you linked, with my credentials, ie client id, domain, and client id

And im still getting the same issue.

**Reply by exhaledev** [reply]
hey @shahbaz17 

> > Hey @exhaledev
> > Can you share the name of the verifier?
> > Also, what are you passing in clientId of connectTo?
> 
> Yeah sure, here is a screenshot below <img alt="image" width="571" src="https://user-images.githubusercontent.com/105014422/202238654-93b6843a-5b4f-499a-9dc3-8469d268faf7.png">

**Reply by shahbaz17** [reply]
Hey @exhaledev 

While creating the verifier with Auth0, which `Authentication Type` have you selected?

**Reply by exhaledev** [reply]
> Hey @exhaledev
> 
> While creating the verifier with Auth0, which `Authentication Type` have you selected?

@shahbaz17 
I used Regular Web Application

**Reply by shahbaz17** [reply]
Also, can you try passing `"name"` instead of `"sub"` as `verifierIdField` inside extraLoginOptions?

**Reply by shahbaz17** [reply]
<img width="734" alt="Screenshot 2022-11-16 at 10 23 57 PM" src="https://user-images.githubusercontent.com/6962565/202243704-7e46f360-91bb-4df4-82eb-ab0810d0f832.png">

I mean, from these, what have you selected?

**Reply by exhaledev** [reply]
Oh, sorry I got confused, I used email passwordless.

"Also, can you try passing "name" instead of "sub" as verifierIdField inside extraLoginOptions?"

That made it work . Changing from sub to name

**Reply by shahbaz17** [reply]
Yes, that will work because selecting `email passwordless` as `Authentication Type` sets the `name` as the verifierIdField. And you were passing `sub` which is set-up when `Google` or any other social providers are selected.

**Comment by shahbaz17** [comment]
When creating a Verifier with Auth0 for `Email Passwordless`, one has to use `"name"` as `verifierIdField` inside connectTo():

```js
import { WALLET_ADAPTERS, CHAIN_NAMESPACES } from "@web3auth/base";

await web3auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
  loginProvider: "jwt",
  extraLoginOptions: {
    domain: "https://YOUR-AUTH0-DOMAIN", // Please append "https://" before your domain
    verifierIdField: "name",
  },
});
```

**Reply by exhaledev** [reply]
Awesome ! Thank you so much !! Very handy fix

**Reply by shahbaz17** [reply]
I have updated [the auth0 guide](https://web3auth.io/docs/guides/auth0#login) to reflect this change.

Thanks for bringing this to our attention.
